app/src/main/java/com/cornellappdev/resell/android/viewmodel/SearchViewModel.kt

```diff
@@ -43,23 +43,26 @@ abstract class SearchViewModel(
 
     fun onQueryChanged(query: String) {
         applyMutation { copy(query = query) }
-
-        // Try to fetch listings from backend
-        viewModelScope.launch {
-            applyMutation {
-                copy(
-                    listings = ResellApiResponse.Pending
-                )
-            }
-            try {
-                val response = searchRepository.searchPostByUser(stateValue().uid, query)
-                // By this time, the query may have changed. If so, ignore.
-                if (stateValue().query == query) {
-                    applyMutation { copy(listings = ResellApiResponse.Success(response)) }
+        if (query.isEmpty() || query.isBlank()) {
+            applyMutation { copy(listings = ResellApiResponse.Success(emptyList())) }
+        } else {
+            // Try to fetch listings from backend
```

Comment:
Instead of wrapping everything here in an `else`, just to reduce indentation, let's make the `if` body `return`. Then, delete the `else`. The body then will only run if the if does not return.

I think reducing indentation like this when possible is good :-)